pilot/cmd/pilot-agent/main.go

```diff
@@ -483,6 +488,36 @@ func init() {
 	}))
 }
 
+func waitForCerts(fname string, maxWait time.Duration) {
+	log.Infof("waiting %v for certificates to become available", maxWait)
+
+	logDelay := 1 * time.Second
+	nextLog := time.Now().Add(logDelay)
+	endWait := time.Now().Add(maxWait)
+
+	for {
+		_, err := os.Stat(fname)
+		if err == nil {
+			return
+		}
+		if !os.IsNotExist(err) { // another error (e.g., permission) - likely no point in waiting longer
+			log.Errora("error while waiting for certificates:", err.Error())
```

Comment:
I meant `log.Errora("error while waiting for certificates", err)` but that's fine too

Reply:
changed